isb_web/main.py

```diff
@@ -210,6 +211,30 @@ async def login(request: fastapi.Request):
     return await _cli.authorize_redirect(request, OAUTH_REDIRECT_URL, state=state)
 
 
+@app.get("/orcid_token", include_in_schema=False)
+def get_orcid_token(request: fastapi.Request, response: fastapi.Response):
+    """
+    Called by a browser as the redirect URI after successfully authenticating with orcid.  Immediately turns the orcid
+    provided code into an orcid token response, and sets cookies to store the token data.
+
+    Args:
+        request: starlette request instance
+
+    Returns:
+        For now, the raw token payload, though this is likely to change to a ReactJS-backed page of some sort.
+    """
+    code = request.query_params.get("code")
+    token_payload = orcid.exchange_code_for_token(code, requests.session())
+    if token_payload is not None:
+        expires = token_payload.get("expires_in")
+        response.set_cookie(key="access_token", value=token_payload.get("access_token"), expires=expires)
+        response.set_cookie(key="refresh_token", value=token_payload.get("refresh_token"), expires=expires)
+        response.set_cookie(key="orcid", value=token_payload.get("orcid"), expires=expires)
```

Comment:
Should the client-side also need to set cookies?

Reply:
No, I think this is all we need.  I think what we'll end up doing is having this response redirect to a view page you will write.